src/core/stream/period/period_stream.ts

```diff
@@ -181,17 +191,30 @@ export default function PeriodStream({
             EMPTY;
 
           const bufferGarbageCollector$ = garbageCollectors.get(qSourceBuffer);
-          const adaptationStream$ = createAdaptationStream(adaptation, qSourceBuffer);
+          const adaptationStream$ = createAdaptationStream(adaptation,
+                                                           qSourceBuffer).pipe(share());
+
+          const firstPushedSegmentOnAdaptationChange$ = needsBufferFlushing$.pipe(
+            switchMap(() => {
+              return adaptationStream$.pipe(
+                filter((value) => value.type === "added-segment"),
+                map(() => EVENTS.needsSourceBufferFlush(bufferType)),
```

Comment:
After picturing what happens here, there is something that bothers me.

We are awaiting for the first segment of the new Adaptation to be added before doing a little seek around the current position.

This means:
  1. That we will have to wait until that segment is loaded before the "flush" is actually done.

      We could still be playing meanwhile the old added segments (and even if we removed it first from the SourceBuffers, the decoder's internal buffer might not become synchronized with that
      This would mean that depending on the user's bandwidth he/she could wait a long time before the real transition happens.

  2. The added segment could technically be an init segment (in which case it contains no data) or a segment for a totally other place than the one we're currently at.

  3. The old segments from the previous Adaptation are left there. This could mean that if the bandwidth is not fast enough, we would play the old Adaptation instead of the new one.

So in the end, this behavior here make little sense.
Based on this I see two possibilities:

  - either we first remove the segments from the old Adaptation then perform the little seek.
     Here it is unclear what happens when removing segments that are currently playing. some environments might have issues with that. Also this task could technically take time and as such we could still be playing for some time until the removing is done.
      So we might also want to pause the playback temporarly here (e.g. by setting the playbackRate at `0` as a kind of "false buffering" - this may be a little more advanced so if we come to that do not hesitate to ask for help).

  - either we wait that the first segment is pushed to the current position. This would also most likely first need some removing and "false buffering" so it might also be harder to do.

All that might be a little complex, so we could also only add a "reload" mode for now.

Reply:
So, I gave another try here.

I got rid of the previous observable, then I'm using the `getAdaptationSwitchStrategy` to know what I've to remove, then I'm waiting for the buffer to be fully removed, to trigger the `needsSourceBufferFlush`. Finally, I'm performing a small seek.

It seems that the `removeBuffer` method on `SourceBuffer` don't take too long.

Reply:
> It seems that the removeBuffer method on SourceBuffer don't take too long.

How much time?
Even if there's a lot of buffer around?
Also, does it work well on most devices?

If it can reach something like around 100ms or more:
  - it might make sense to save the initial position so we can seek back closer to the step where the user change the audio track
  - We could perhaps save time by first removing around the current position, then seek to flush (to be rebuffering) and then take the time to clean the rest before pushing the new segments (a downside is that it might need a second flush when all is removed)